src/Relation/Nullary.agda

```diff
@@ -23,3 +23,18 @@ infix 3 ¬_
 data Dec {p} (P : Set p) : Set p where
   yes : ( p :   P) → Dec P
   no  : (¬p : ¬ P) → Dec P
+
+
+-- Given an irrelevant proof of a decidable type, a proof can
+-- be recomputed and subsequently used in relevant contexts.
+recompute : ∀ {a} {A : Set a} → Dec A → .A → A
+recompute (yes x) _ = x
+recompute {A = A} (no ¬p) x with ¬-i ¬p x
+  where
+    ⊥-i : ∀ .(_ : ⊥) → ⊥
```

Comment:
This is subsumed by `⊥-elim` in [`Data.Empty.Irrelevant`](https://agda.github.io/agda-stdlib/Data.Empty.Irrelevant.html#291).